streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoinTest.java

```diff
@@ -816,10 +816,12 @@ public void testWindowing() {
         stream1 = builder.stream(topic1, consumed);
         stream2 = builder.stream(topic2, consumed);
 
+        final Duration timeDifference = ofMillis(100L);
+        final Duration grace = ofMillis(104);
```

Comment:
Good idea with the sensor.

Just for me to understand, is 150 an arbitrarily chosen value or how did you come up with it?
Wouldn't it make sense to set it to (max_timestamp in test case - min_timestamp in test case) e.g. 1104 - 899

Reply:
On second thought, note my coment on 'Why + 1 '.

Reply:
Yes, `150` was random -- just a "round" number that is big enough. -- Don't think we need to set a "strict" bound by computing the minimum grace period we need (IMHO, it would just be distracting if one read the test to figure out where `1104 - 899` comes from...